cmake/modules/FindMicroHttpd.cmake

```diff
@@ -11,24 +11,84 @@ if(NOT TARGET ${APP_NAME_LC}::${CMAKE_FIND_PACKAGE_NAME})
 
   include(cmake/scripts/common/ModuleHelpers.cmake)
 
+  macro(buildmacroMicroHttpd)
+
+    if(WIN32 OR WINDOWS_STORE)
+
+      set(patches "${CMAKE_SOURCE_DIR}/tools/depends/target/${${CMAKE_FIND_PACKAGE_NAME}_MODULE_LC}/01-win-cmake.patch")
+
+      generate_patchcommand("${patches}")
+      unset(patches)
+
+      set(${${CMAKE_FIND_PACKAGE_NAME}_MODULE}_DEBUG_POSTFIX _d)
+
+      set(CMAKE_ARGS -DDUMMY_ARGS=ON)
+
+    else()
+      # Todo: gnutls libgcrypt libgpg-error
+      # find_package(xxx)
+
+      if (CMAKE_HOST_SYSTEM_NAME MATCHES "(Free|Net|Open)BSD")
+        find_program(MAKE_EXECUTABLE gmake)
+      endif()
+      find_program(MAKE_EXECUTABLE make REQUIRED)
+
+      if(CMAKE_SYSTEM_NAME MATCHES "Darwin")
+        # blanket disable timespec_get use for apple platforms. timespec_get was introduced in
+        # __API_AVAILABLE(macosx(10.15), ios(13.0), tvos(13.0), watchos(6.0)) but older platforms
+        # are failing to run.
+        set(EXTRA_ARGS mhd_cv_func_timespec_get=no)
+      endif()
+
+      set(CONFIGURE_COMMAND ./configure --prefix ${DEPENDS_PATH}
+                                        --disable-shared
+                                        --disable-doc
+                                        --disable-examples
+                                        --disable-curl
+                                        --enable-https
+                                        ${EXTRA_ARGS})
+
+      set(BUILD_COMMAND ${MAKE_EXECUTABLE})
+      set(INSTALL_COMMAND ${MAKE_EXECUTABLE} install)
+
+      set(BUILD_IN_SOURCE 1)
+
+    endif()
+
+    BUILD_DEP_TARGET()
+  endmacro()
+
   set(${CMAKE_FIND_PACKAGE_NAME}_MODULE_LC libmicrohttpd)
-  set(${${CMAKE_FIND_PACKAGE_NAME}_MODULE_LC}_DISABLE_VERSION ON)
 
   SETUP_BUILD_VARS()
 
   SETUP_FIND_SPECS()
 
   SEARCH_EXISTING_PACKAGES()
 
+  if(("${${${CMAKE_FIND_PACKAGE_NAME}_SEARCH_NAME}_VERSION}" VERSION_LESS ${${${CMAKE_FIND_PACKAGE_NAME}_MODULE}_VER} AND ENABLE_INTERNAL_LIBMICROHTTPD) OR
+     ((CORE_SYSTEM_NAME STREQUAL linux OR CORE_SYSTEM_NAME STREQUAL freebsd) AND ENABLE_INTERNAL_LIBMICROHTTPD))
```

Comment:
I have noticed that this also triggers a rebuild on webOS because `CORE_SYSTEM_NAME` is linux on webOS and `ENABLE_INTERNAL_LIBMICROHTTPD` is true so the 2nd condition applies whilst it should only be the first one. Should the 2nd condition also check if it is not a `KODI_DEPENDSBUILD`?